class/class7.1/src/MaybeTheBestCodeEver.cpp

```diff
@@ -0,0 +1,77 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <limits>
+#include "Rectangle.hpp"
+
+class RectangleProcessor {
+public:
+    void addRectangle(double width, double height) {
+        rectangles_.emplace_back(width, height);
```

Comment:
minor: Try to check return value of emplace_back call
auto& rect = rectangles_.emplace_back(width, height);
 updateBiggestAndSmallestArea(rect.getArea());